### PR TITLE
chore(output): refactor

### PR DESCRIPTION
**Pull request by brendongo**
PR checklist:
- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [ ] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by brendongo** [comment]
output.py still messy but these are the minimum changes necessary to support baseline scanning

File: semgrep/semgrep/commands/scan.py [hunk 1]
```diff
@@ -672,36 +672,57 @@ def scan(
...
+                    disable_nosem=(not enable_nosem),
+                    no_git_ignore=(not use_git_ignore),
+                    timeout=timeout,
+                    max_memory=max_memory,
+                    timeout_threshold=timeout_threshold,
+                    skip_unknown_extensions=(not scan_unknown_extensions),
+                    severity=severity,
+                    optimizations=optimizations,
+                )
+            except SemgrepError as e:
```
Comment (emjin) [inline semgrep/semgrep/commands/scan.py:793]: Yaaaaaaaay

**Review by emjin — APPROVED** [review]
Ooh definitely an improvement

File: semgrep/semgrep/output.py [hunk 2]
```diff
@@ -406,9 +367,15 @@ def build_output(
                 self.profiler,
             )
         if self.settings.output_format == OutputFormat.TEXT:
-            extra["color_output"] = color_output
-            extra["per_finding_max_lines_limit"] = per_finding_max_lines_limit
-            extra["per_line_max_chars_limit"] = per_line_max_chars_limit
+            extra["color_output"] = (
+                self.settings.output_destination is None and self.stdout.isatty(),
```
Comment (emjin) [inline semgrep/semgrep/output.py:371]: Can you add an explanation for this?

**Pull request merged**